demo/src/examples/7 Events.vue

```diff
@@ -0,0 +1,88 @@
+<template>
+  <app-sidebar>
+    <template #title>{{ name }}</template>
+
+    <p>
+      This example listens for all the available events and logs them below. Interact with the map to see the emitted
+      events.
+    </p>
+
+    <ol>
+      <!-- eslint-disable-next-line -->
+      <li v-for="message in messages" :key="message" v-html="message" />
+    </ol>
+
+    <button v-if="messages.length" @click="messages = []">Clear</button>
+  </app-sidebar>
+
+  <div ref="mapRef" class="shadow-xl" />
+</template>
+
+<script setup lang="ts">
+  import { onMounted, ref } from "vue";
+  import { useRoute } from "vue-router";
+  import AppSidebar from "../components/AppSidebar.vue";
+  import maplibregl from "maplibre-gl";
+  import "maplibre-gl/dist/maplibre-gl.css";
+  import style from "../assets/map/style/style.json?url";
+  import MaplibreGlDirections from "@maplibre/maplibre-gl-directions";
+
+  const name = ref(useRoute().matched[0].name);
+
+  const directions = ref<MaplibreGlDirections>();
+
+  const messages = ref<string[]>([]);
+
+  const mapRef = ref();
+  onMounted(() => {
+    const map = new maplibregl.Map({
+      container: mapRef.value,
+      style,
+      center: [-74.1197632, 40.6974034],
+      zoom: 11,
+      customAttribution: "<a href='http://project-osrm.org/' target='_blank'>&copy; OSRM</a>",
+    });
+
+    map.on("load", () => {
+      directions.value = new MaplibreGlDirections(map, {
+        requestOptions: {
+          alternatives: "true",
+        },
+      });
+
+      directions.value.interactive = true;
+
+      directions.value?.on("addwaypoint", (e) => {
```

Comment:
What does the question mark do?

Reply:
Good point :+1:, it's an artifact of copying the code around. Thanks for noticing!